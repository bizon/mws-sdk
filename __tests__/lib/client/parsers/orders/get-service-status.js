const parseXml = require('../../../../../lib/client/parsers')
const parseGetServiceStatusResponse = require('../../../../../lib/client/parsers/base/get-service-status-response')

describe('lib.client.parsers.base.get-service-status-response', () => {
  it('should parse the GetServiceStatusResponse example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <GetServiceStatusResponse xmlns="https://mws.amazonservices.com/Orders/2013-09-01">
        <GetServiceStatusResult>
          <Status>GREEN</Status>
          <Timestamp>2013-09-05T18%3A12%3A21</Timestamp>
        </GetServiceStatusResult>
        <ResponseMetadata>
          <RequestId>d80c6c7b-f7c7-4fa7-bdd7-854711cb3bcc</RequestId>
        </ResponseMetadata>
      </GetServiceStatusResponse>`,
    )

    const res = parseGetServiceStatusResponse('/orders:GetServiceStatusResponse', doc)

    expect(res).toMatchSnapshot()
  })
})
