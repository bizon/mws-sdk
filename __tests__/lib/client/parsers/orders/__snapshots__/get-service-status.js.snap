// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.parsers.base.get-service-status-response should parse the GetServiceStatusResponse example response from MWS doc 1`] = `
Object {
  "getServiceStatusResult": Object {
    "status": "GREEN",
    "timestamp": 2013-09-04T22:00:00.000Z,
  },
  "responseMetadata": Object {
    "requestId": "d80c6c7b-f7c7-4fa7-bdd7-854711cb3bcc",
  },
}
`;
